basalt/nn/optim.mojo

```diff
@@ -22,8 +22,12 @@ fn get_trainable_parameters(g: Graph) -> List[Symbol]:
 
 struct Adam[
     g: Graph,
+    mutability: __mlir_type.i1,
+    lifetime: AnyLifetime[mutability].type,
     trainable_parameters: List[Symbol] = get_trainable_parameters(g),
 ]:
+    var parameters: Reference[Parameters, mutability, lifetime]
```

Comment:
Okay so it is possible right now to have references in a struct (i just didn't know how it seems)